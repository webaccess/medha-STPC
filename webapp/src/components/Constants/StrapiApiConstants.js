export const STRAPI_LOGIN_PATH = "auth/local";
export const STRAPI_DB_URL = "http://localhost:1337/";
