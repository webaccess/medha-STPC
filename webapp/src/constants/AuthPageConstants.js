/* Sign In Constants */
export const SIGN_IN_HEADER = "Sign in";
export const FORGOT_PASSWORD_ROUTE_TEXT = "Forgot password?";
export const NEW_REG_ROUTE_TEXT = "New Registration";
export const SIGN_IN_BUTTON = "Sign in";
export const INVALID_USER = "Invalid email or password";
export const REGISTER = "Register";
export const CREATE_ACTIVITY = "Create Activity";
/* Forgot Password Constants */
export const FORGOT_PASSWORD_HEADER = "Forgot your password?";
export const CONFIRM_NEW_PASS_ALERT = "Please enter new password";
export const OTP_ALERT = "Please enter the OTP sent to";
export const MOBILE_NUMBER_ALERT =
  "Enter the mobile number associated with your account and we'll send you an OTP";
export const SEND_OTP_BUTTON = "Send OTP";
export const INVALID_OTP = "Invalid OTP";
export const BACK_TO_LOGIN_TEXT = "Take me back to login page";
export const PASS_SAVE_BUTTON = "Save Password";
export const RESET_PASS_BUTTON = "Reset Password";
export const VERIFY_OTP_BUTTON = "Verify OTP";
export const RESEND_OTP_BUTTON = "Resend OTP";
export const FORM_TYPE_ENTER_MOBILE = "verify_mobile";
export const FORM_TYPE_VERIFY_OTP = "verify_otp";
export const FORM_TYPE_RESEND_OTP = "resend_otp";
export const FORM_TYPE_CHANGE_PASS = "change_pass";
export const CHANGE_CONTACT_NUMBER = "Change Contact Number";
