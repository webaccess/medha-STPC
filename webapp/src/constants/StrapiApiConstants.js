export const STRAPI_LOGIN_PATH = "auth/local";
export const STRAPI_DB_URL = "http://localhost:1337/";
export const STRAPI_DB_URL_WITHOUT_HASH = "http://localhost:1337";
export const STRAPI_ZONES = "zones";
export const STRAPI_STATES = "states";
export const STRAPI_RPCS = "rpcs";
export const STRAPI_COLLEGES = "colleges";
export const STRAPI_STREAMS = "streams";
export const STRAPI_USERS = "users";
export const STRAPI_DISTRICTS = "districts";
export const STRAPI_ROLES = "users-permissions/roles";
export const STRAPI_REQUEST_STUDENT_OTP = "otps/requestotpforstudent";
export const STRAPI_CHECK_OTP = "otps/checkotp";
export const STRAPI_REGISTER_STUDENT = "students/register";
export const STRAPI_STUDENT = "students";
export const STRAPI_UPDATE = "update";
export const STRAPI_EDUCATIONS = "educations";
export const STRAPI_REQUEST_OTP = "otps/requestotp";
export const STRAPI_VALIDATE_OTP = "otps/validateotp";
export const STRAPI_CHANGE_PASS_URL = "auth/reset-password";
export const STRAPI_STUDENTS = "students";
export const STRAPI_UPLOAD = "upload";
export const STRAPI_DELETE_UPLOAD = "file";
export const STRAPI_ACADEMIC_YEARS = "academic-years";
export const STRAPI_ACADEMIC_HISTORY = "academic-histories";
export const STRAPI_EVENTS = "events";
export const STRAPI_ACTIVITY = "activities";
export const STRAPI_COLLEGE_ACTIVITY = "activity";
