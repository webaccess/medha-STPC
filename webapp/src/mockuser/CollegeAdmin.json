{
  "jwt": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MiwiaWF0IjoxNTk0MjMzMzc0LCJleHAiOjE1OTY4MjUzNzR9.fvVawPPhOtYOKVNXj4rlVVv8t1c1xzB8DRYWW-XY35A",
  "user": {
    "id": 2,
    "username": "7710950932",
    "email": "7710950932@gmail.com",
    "provider": "local",
    "confirmed": null,
    "blocked": false,
    "role": {
      "id": 6,
      "name": "College Admin",
      "description": "College Admin",
      "type": "College Admin"
    },
    "state": {
      "id": 1,
      "name": "Uttar Pradesh",
      "is_active": true,
      "abbreviation": "UP",
      "identifier": "",
      "country": 1,
      "created_at": "2020-06-24T07:09:28.251Z",
      "updated_at": "2020-06-24T07:09:28.251Z"
    },
    "zone": {
      "id": 1,
      "name": "West Zone - Daurala (Meerut)",
      "state": 1,
      "created_at": "2020-06-24T07:09:35.321Z",
      "updated_at": "2020-06-24T07:09:35.321Z"
    },
    "rpc": {
      "id": 1,
      "name": "Agra",
      "state": 1,
      "main_college": 3,
      "created_at": "2020-06-24T07:09:35.350Z",
      "updated_at": "2020-06-24T08:50:17.773Z"
    },
    "contact": {
      "id": 4,
      "name": "College Admin",
      "phone": "7710950932",
      "phone_other": null,
      "email": "7710950932@gmail.com",
      "email_other": null,
      "address_1": null,
      "address_2": null,
      "city": null,
      "pincode": null,
      "contact_type": "individual",
      "organization": null,
      "country": null,
      "village": null,
      "state": 1,
      "district": null,
      "individual": 2,
      "user": 2,
      "created_at": "2020-06-24T07:20:32.867Z",
      "updated_at": "2020-06-24T07:20:32.867Z"
    },
    "created_at": "2020-06-24T07:20:32.864Z",
    "updated_at": "2020-06-24T07:20:32.869Z",
    "rpcAdminRole": {
      "id": 5,
      "name": "RPC Admin",
      "description": "RPC Admin",
      "type": "RPC Admin"
    },
    "studentInfo": {
      "id": 2,
      "first_name": "College",
      "last_name": "Admin",
      "contact": {
        "id": 4,
        "name": "College Admin",
        "phone": "7710950932",
        "phone_other": null,
        "email": "7710950932@gmail.com",
        "email_other": null,
        "address_1": null,
        "address_2": null,
        "city": null,
        "pincode": null,
        "contact_type": "individual",
        "organization": null,
        "country": null,
        "village": null,
        "state": 1,
        "district": null,
        "individual": 2,
        "user": 2,
        "created_at": "2020-06-24T07:20:32.867Z",
        "updated_at": "2020-06-24T07:20:32.867Z"
      },
      "stream": null,
      "is_verified": null,
      "is_physically_challenged": null,
      "date_of_birth": null,
      "gender": null,
      "roll_number": null,
      "organization": {
        "id": 1,
        "name": "Government Polytechnic, Mainpuri",
        "contact": {
          "id": 2,
          "name": "Government Polytechnic, Mainpuri",
          "phone": "0356273400",
          "phone_other": null,
          "email": "collegeemail1@gmail.com",
          "email_other": null,
          "address_1": "UP",
          "address_2": null,
          "city": null,
          "pincode": null,
          "contact_type": "organization",
          "organization": 1,
          "country": null,
          "village": null,
          "state": {
            "id": 1,
            "name": "Uttar Pradesh",
            "is_active": true,
            "abbreviation": "UP",
            "identifier": "",
            "country": 1,
            "created_at": "2020-06-24T07:09:28.251Z",
            "updated_at": "2020-06-24T07:09:28.251Z"
          },
          "district": {
            "id": 49,
            "name": "Mainpuri",
            "is_active": true,
            "abbreviation": "MP",
            "identifier": null,
            "state": 1,
            "created_at": "2020-06-24T07:09:28.554Z",
            "updated_at": "2020-06-24T07:09:28.554Z"
          },
          "individual": null,
          "user": null,
          "created_at": "2020-06-24T07:18:46.694Z",
          "updated_at": "2020-06-24T07:18:46.694Z"
        },
        "zone": {
          "id": 1,
          "name": "West Zone - Daurala (Meerut)",
          "state": 1,
          "created_at": "2020-06-24T07:09:35.321Z",
          "updated_at": "2020-06-24T07:09:35.321Z"
        },
        "rpc": {
          "id": 1,
          "name": "Agra",
          "state": 1,
          "main_college": 3,
          "created_at": "2020-06-24T07:09:35.350Z",
          "updated_at": "2020-06-24T08:50:17.773Z"
        },
        "college_code": "GPM23",
        "principal": null,
        "is_blocked": false,
        "created_at": "2020-06-24T07:18:46.679Z",
        "updated_at": "2020-06-24T07:18:46.697Z",
        "stream_strength": [
          {
            "id": 1,
            "stream": {
              "id": 2,
              "name": "Computer Science And Engineering",
              "created_at": "2020-06-24T07:09:35.414Z",
              "updated_at": "2020-06-24T07:09:35.414Z"
            },
            "first_year_strength": 0,
            "second_year_strength": 0,
            "third_year_strength": 0
          },
          {
            "id": 2,
            "stream": {
              "id": 1,
              "name": "Mechanical Engineering (Production)",
              "created_at": "2020-06-24T07:09:35.408Z",
              "updated_at": "2020-06-24T07:09:35.408Z"
            },
            "first_year_strength": 0,
            "second_year_strength": 0,
            "third_year_strength": 0
          },
          {
            "id": 3,
            "stream": {
              "id": 3,
              "name": "Electronics Engineering",
              "created_at": "2020-06-24T07:09:35.419Z",
              "updated_at": "2020-06-24T07:09:35.419Z"
            },
            "first_year_strength": 0,
            "second_year_strength": 0,
            "third_year_strength": 0
          }
        ],
        "tpos": []
      },
      "father_full_name": null,
      "mother_full_name": null,
      "middle_name": null,
      "created_at": "2020-06-24T07:20:32.866Z",
      "updated_at": "2020-06-24T07:20:32.871Z",
      "profile_photo": null,
      "future_aspirations": []
    },
    "first_name": "College",
    "last_name": "Admin"
  }
}
