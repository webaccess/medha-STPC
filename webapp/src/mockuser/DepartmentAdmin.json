{
  "jwt": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MSwiaWF0IjoxNTkzODc4NDk0LCJleHAiOjE1OTY0NzA0OTR9.v30khS2ujqelQIAwyWLaVHzwzFu_GbJa1QNbppp2jFM",
  "user": {
    "id": 1,
    "username": "9029161582",
    "email": "9029161582@gmail.com",
    "provider": "local",
    "confirmed": null,
    "blocked": false,
    "role": {
      "id": 8,
      "name": "Department Admin",
      "description": "Department Admin",
      "type": "Department Admin"
    },
    "state": null,
    "zone": null,
    "rpc": null,
    "contact": {
      "id": 8,
      "name": "Department Admin",
      "phone": "9029161582",
      "phone_other": null,
      "email": "9029161582@gmail.com",
      "email_other": null,
      "address_1": "UP",
      "address_2": null,
      "city": null,
      "pincode": null,
      "contact_type": "individual",
      "organization": null,
      "country": null,
      "village": null,
      "state": 1,
      "district": null,
      "individual": 1,
      "user": 1,
      "created_at": "2020-06-25T07:08:46.889Z",
      "updated_at": "2020-06-25T16:05:06.728Z"
    },
    "created_at": "2020-06-25T07:08:46.889Z",
    "updated_at": "2020-06-25T16:05:06.734Z",
    "rpcAdminRole": {
      "id": 5,
      "name": "RPC Admin",
      "description": "RPC Admin",
      "type": "RPC Admin"
    },
    "studentInfo": {
      "id": 1,
      "first_name": "Department",
      "last_name": "Admin",
      "contact": {
        "id": 8,
        "name": "Department Admin",
        "phone": "9029161582",
        "phone_other": null,
        "email": "9029161582@gmail.com",
        "email_other": null,
        "address_1": "UP",
        "address_2": null,
        "city": null,
        "pincode": null,
        "contact_type": "individual",
        "organization": null,
        "country": null,
        "village": null,
        "state": 1,
        "district": null,
        "individual": 1,
        "user": 1,
        "created_at": "2020-06-25T07:08:46.889Z",
        "updated_at": "2020-06-25T16:05:06.728Z"
      },
      "stream": null,
      "is_verified": true,
      "is_physically_challenged": false,
      "date_of_birth": "1999-12-31T18:30:00.000Z",
      "gender": null,
      "roll_number": null,
      "organization": null,
      "father_full_name": "",
      "mother_full_name": "",
      "middle_name": " ",
      "created_at": "2020-06-25T07:08:46.889Z",
      "updated_at": "2020-06-25T07:08:46.889Z",
      "profile_photo": null,
      "future_aspirations": []
    },
    "first_name": "Department",
    "last_name": "Admin"
  }
}
