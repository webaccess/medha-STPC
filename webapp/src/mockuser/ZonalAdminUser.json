{
  "jwt": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6NDAsImlhdCI6MTU5NDIwMjU5OCwiZXhwIjoxNTk2Nzk0NTk4fQ.fZdrk2PDiuOElrJ0FW5qdTvcDRvixuyDPfMh-tpO2qo",
  "user": {
      "id": 40,
      "username": "9029161583",
      "email": "9029161583@gmail.com",
      "provider": "local",
      "confirmed": null,
      "blocked": false,
      "role": {
          "id": 4,
          "name": "Zonal Admin",
          "description": "Zonal Admin",
          "type": "Zonal Admin"
      },
      "state": {
          "id": 1,
          "name": "Uttar Pradesh",
          "is_active": true,
          "abbreviation": "UP",
          "identifier": "",
          "country": 1,
          "created_at": "2020-06-24T07:09:28.251Z",
          "updated_at": "2020-06-24T07:09:28.251Z"
      },
      "zone": {
          "id": 1,
          "name": "West Zone - Daurala (Meerut)",
          "state": 1,
          "created_at": "2020-06-24T07:09:35.321Z",
          "updated_at": "2020-06-24T07:09:35.321Z"
      },
      "rpc": null,
      "contact": {
          "id": 42,
          "name": "Zonal Admin",
          "phone": "9029161583",
          "phone_other": null,
          "email": "9029161583@gmail.com",
          "email_other": null,
          "address_1": null,
          "address_2": null,
          "city": null,
          "pincode": null,
          "contact_type": "individual",
          "organization": null,
          "country": null,
          "village": null,
          "state": 1,
          "district": null,
          "individual": 40,
          "user": 40,
          "created_at": "2020-06-24T08:53:11.954Z",
          "updated_at": "2020-06-24T08:53:11.954Z"
      },
      "created_at": "2020-06-24T08:53:11.952Z",
      "updated_at": "2020-06-24T08:53:11.955Z",
      "rpcAdminRole": {
          "id": 5,
          "name": "RPC Admin",
          "description": "RPC Admin",
          "type": "RPC Admin"
      },
      "studentInfo": {
          "id": 40,
          "first_name": "Zonal",
          "last_name": "Admin",
          "contact": {
              "id": 42,
              "name": "Zonal Admin",
              "phone": "9029161583",
              "phone_other": null,
              "email": "9029161583@gmail.com",
              "email_other": null,
              "address_1": null,
              "address_2": null,
              "city": null,
              "pincode": null,
              "contact_type": "individual",
              "organization": null,
              "country": null,
              "village": null,
              "state": 1,
              "district": null,
              "individual": 40,
              "user": 40,
              "created_at": "2020-06-24T08:53:11.954Z",
              "updated_at": "2020-06-24T08:53:11.954Z"
          },
          "stream": null,
          "is_verified": null,
          "is_physically_challenged": null,
          "date_of_birth": null,
          "gender": null,
          "roll_number": null,
          "organization": null,
          "father_full_name": null,
          "mother_full_name": null,
          "middle_name": null,
          "created_at": "2020-06-24T08:53:11.953Z",
          "updated_at": "2020-06-24T08:53:11.957Z",
          "profile_photo": null,
          "future_aspirations": []
      },
      "first_name": "Zonal",
      "last_name": "Admin"
  }
}