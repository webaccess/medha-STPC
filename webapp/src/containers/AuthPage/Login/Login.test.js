import React from "react";
import { shallow, mount } from "enzyme";
import LogIn from "./Login.js";
import axios from "axios";
import * as strapiApiConstants from "../../../constants/StrapiApiConstants";
import * as medhaAdminUser from "../../../mockuser/MedhaAdmin.json";
import * as collegeAdminUser from "../../../mockuser/CollegeAdmin.json";
import * as StudentUser from "../../../mockuser/StudentUser.json";
import * as ZonalUser from "../../../mockuser/ZonalAdminUser.json";
import * as DepartmentAdmin from "../../../mockuser/DepartmentAdmin.json";

React.useLayoutEffect = React.useEffect;

jest.mock("axios");
jest.mock("react-router", () => ({
  ...jest.requireActual("react-router"),
  useHistory: () => ({ push: jest.fn() })
}));

const originalConsoleError = console.error;

beforeEach(() => {
  console.error = jest.fn();
});

afterEach(() => {
  console.error = originalConsoleError;
});

const simulateChangeOnInput = (wrapper, inputSelector, newValue, inputName) => {
  const input = wrapper.find(inputSelector);

  input.simulate("change", {
    persist: jest.fn(),
    target: { name: inputName, value: newValue }
  });
  return wrapper.find(inputSelector);
};

describe("Login", () => {
  it("should match the snapshot", () => {
    const wrapper = shallow(<LogIn />);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("Should check contact number has value", () => {
    const wrapper = shallow(<LogIn />);
    const contactInput = simulateChangeOnInput(
      wrapper,
      "#contactnumber",
      "9029161582",
      "identifier"
    );

    expect(contactInput.props().value).toBe("9029161582");
  });

  it("Should check password has value", () => {
    const wrapper = shallow(<LogIn />);
    const contactInput = simulateChangeOnInput(
      wrapper,
      "#password",
      "admin1234",
      "password"
    );

    expect(contactInput.props().value).toBe("admin1234");
  });

  it("Should check post request for Sign In for Medha Admin", async () => {
    const wrapper = shallow(<LogIn />);
    const data = medhaAdminUser;
    const postSpy = jest.spyOn(axios, "post").mockImplementation(() => {
      return new Promise(resolve => {
        return resolve({ data: data });
      });
    });

    /**Setting the identifier and password */
    simulateChangeOnInput(
      wrapper,
      "#contactnumber",
      "9029161582",
      "identifier"
    );
    simulateChangeOnInput(wrapper, "#password", "admin1234", "password");

    /** Mock data which is expected after setting username and password */
    const mockDataRequest = {
      identifier: "9029161582",
      password: "admin1234"
    };

    /** This simulates the Onsubmit event */
    wrapper.find("#form").simulate("submit", {
      preventDefault: jest.fn()
    });

    /** This actually checks whether the post method is called */
    expect(postSpy).toBeCalled();

    /** This actually checks whether the post method is called with proper url and request body */
    expect(postSpy).toBeCalledWith(
      `${
        strapiApiConstants.STRAPI_DB_URL + strapiApiConstants.STRAPI_LOGIN_PATH
      }`,
      expect.objectContaining(mockDataRequest)
    );

    /** postSpy.mock.results.pop().value this gets the promise of the axios request */
    const postPromise = postSpy.mock.results.pop().value;

    /** Check promise */
    return postPromise.then(postResponse => {
      expect(postResponse).toStrictEqual({ data: data });
    });
  });

  it("Should check post request for Sign In for Medha Admin", async () => {
    const wrapper = shallow(<LogIn />);
    const data = collegeAdminUser;
    const postSpy = jest.spyOn(axios, "post").mockImplementation(() => {
      return new Promise(resolve => {
        return resolve({ data: data });
      });
    });

    /**Setting the identifier and password */
    simulateChangeOnInput(
      wrapper,
      "#contactnumber",
      "9029161582",
      "identifier"
    );
    simulateChangeOnInput(wrapper, "#password", "admin1234", "password");

    /** Mock data which is expected after setting username and password */
    const mockDataRequest = {
      identifier: "9029161582",
      password: "admin1234"
    };

    /** This simulates the Onsubmit event */
    wrapper.find("#form").simulate("submit", {
      preventDefault: jest.fn()
    });

    /** This actually checks whether the post method is called */
    expect(postSpy).toBeCalled();

    /** This actually checks whether the post method is called with proper url and request body */
    expect(postSpy).toBeCalledWith(
      `${
        strapiApiConstants.STRAPI_DB_URL + strapiApiConstants.STRAPI_LOGIN_PATH
      }`,
      expect.objectContaining(mockDataRequest)
    );

    /** postSpy.mock.results.pop().value this gets the promise of the axios request */
    const postPromise = postSpy.mock.results.pop().value;

    /** Check promise */
    return postPromise.then(postResponse => {
      expect(postResponse).toStrictEqual({ data: data });
    });
  });

  it("Should check post request for Sign In for Medha Admin", async () => {
    const wrapper = shallow(<LogIn />);
    const data = StudentUser;
    const postSpy = jest.spyOn(axios, "post").mockImplementation(() => {
      return new Promise(resolve => {
        return resolve({ data: data });
      });
    });

    /**Setting the identifier and password */
    simulateChangeOnInput(
      wrapper,
      "#contactnumber",
      "9029161582",
      "identifier"
    );
    simulateChangeOnInput(wrapper, "#password", "admin1234", "password");

    /** Mock data which is expected after setting username and password */
    const mockDataRequest = {
      identifier: "9029161582",
      password: "admin1234"
    };

    /** This simulates the Onsubmit event */
    wrapper.find("#form").simulate("submit", {
      preventDefault: jest.fn()
    });

    /** This actually checks whether the post method is called */
    expect(postSpy).toBeCalled();

    /** This actually checks whether the post method is called with proper url and request body */
    expect(postSpy).toBeCalledWith(
      `${
        strapiApiConstants.STRAPI_DB_URL + strapiApiConstants.STRAPI_LOGIN_PATH
      }`,
      expect.objectContaining(mockDataRequest)
    );

    /** postSpy.mock.results.pop().value this gets the promise of the axios request */
    const postPromise = postSpy.mock.results.pop().value;

    /** Check promise */
    return postPromise.then(postResponse => {
      expect(postResponse).toStrictEqual({ data: data });
    });
  });

  it("Should check post request for Sign In for Zonal Admin", async () => {
    const wrapper = shallow(<LogIn />);
    const data = ZonalUser;
    const postSpy = jest.spyOn(axios, "post").mockImplementation(() => {
      return new Promise(resolve => {
        return resolve({ data: data });
      });
    });

    /**Setting the identifier and password */
    simulateChangeOnInput(
      wrapper,
      "#contactnumber",
      "9029161582",
      "identifier"
    );
    simulateChangeOnInput(wrapper, "#password", "admin1234", "password");

    /** Mock data which is expected after setting username and password */
    const mockDataRequest = {
      identifier: "9029161582",
      password: "admin1234"
    };

    /** This simulates the Onsubmit event */
    wrapper.find("#form").simulate("submit", {
      preventDefault: jest.fn()
    });

    /** This actually checks whether the post method is called */
    expect(postSpy).toBeCalled();

    /** This actually checks whether the post method is called with proper url and request body */
    expect(postSpy).toBeCalledWith(
      `${
        strapiApiConstants.STRAPI_DB_URL + strapiApiConstants.STRAPI_LOGIN_PATH
      }`,
      expect.objectContaining(mockDataRequest)
    );

    /** postSpy.mock.results.pop().value this gets the promise of the axios request */
    const postPromise = postSpy.mock.results.pop().value;

    /** Check promise */
    return postPromise.then(postResponse => {
      expect(postResponse).toStrictEqual({ data: data });
    });
  });

  it("Should check post request for Sign In for Zonal Admin", async () => {
    const wrapper = shallow(<LogIn />);
    const data = DepartmentAdmin;
    const postSpy = jest.spyOn(axios, "post").mockImplementation(() => {
      return new Promise(resolve => {
        return resolve({ data: data });
      });
    });

    /**Setting the identifier and password */
    simulateChangeOnInput(
      wrapper,
      "#contactnumber",
      "9029161582",
      "identifier"
    );
    simulateChangeOnInput(wrapper, "#password", "admin1234", "password");

    /** Mock data which is expected after setting username and password */
    const mockDataRequest = {
      identifier: "9029161582",
      password: "admin1234"
    };

    /** This simulates the Onsubmit event */
    wrapper.find("#form").simulate("submit", {
      preventDefault: jest.fn()
    });

    /** This actually checks whether the post method is called */
    expect(postSpy).toBeCalled();

    /** This actually checks whether the post method is called with proper url and request body */
    expect(postSpy).toBeCalledWith(
      `${
        strapiApiConstants.STRAPI_DB_URL + strapiApiConstants.STRAPI_LOGIN_PATH
      }`,
      expect.objectContaining(mockDataRequest)
    );

    /** postSpy.mock.results.pop().value this gets the promise of the axios request */
    const postPromise = postSpy.mock.results.pop().value;

    /** Check promise */
    return postPromise.then(postResponse => {
      expect(postResponse).toStrictEqual({ data: data });
    });
  });
});
