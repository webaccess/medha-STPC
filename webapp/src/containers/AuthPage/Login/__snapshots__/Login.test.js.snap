// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match the snapshot 1`] = `"<div class=\\"makeStyles-masterlogin2-7\\"><div class=\\"makeStyles-masterlogin1-8\\"><div class=\\"makeStyles-masterlogin-9\\"><div class=\\"MuiPaper-root makeStyles-rootDesktop-5 MuiPaper-elevation1 MuiPaper-rounded\\"><div class=\\"MuiCardContent-root\\"><div class=\\"makeStyles-paper-1\\"><div class=\\"makeStyles-signin_header-11\\"><div class=\\"makeStyles-loginlock-10\\"><div class=\\"makeStyles-cardIcon-52\\"><span class=\\"material-icons MuiIcon-root\\" aria-hidden=\\"true\\">lock</span></div></div><h1 class=\\"MuiTypography-root MuiTypography-h5\\" style=\\"font-weight:700\\">Sign in</h1></div><form novalidate=\\"\\" id=\\"form\\"><div class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined\\" data-shrink=\\"false\\" for=\\"contactnumber\\" id=\\"contactnumber-label\\">Contact Number</label><div class=\\"MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl\\"><input type=\\"text\\" aria-invalid=\\"false\\" autofocus=\\"\\" id=\\"contactnumber\\" name=\\"identifier\\" value=\\"\\" class=\\"MuiInputBase-input MuiOutlinedInput-input\\"/><fieldset aria-hidden=\\"true\\" class=\\"PrivateNotchedOutline-root-156 MuiOutlinedInput-notchedOutline\\"><legend class=\\"PrivateNotchedOutline-legendLabelled-158\\"><span>Contact Number</span></legend></fieldset></div></div><div class=\\"MuiFormControl-root makeStyles-margin-3 MuiFormControl-fullWidth\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined\\" data-shrink=\\"false\\" for=\\"outlined-adornment-password\\">Password</label><div class=\\"MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd\\" InputLabelProps=\\"[object Object]\\" InputProps=\\"[object Object]\\"><input type=\\"password\\" aria-invalid=\\"false\\" id=\\"password\\" name=\\"password\\" value=\\"\\" class=\\"MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd\\"/><div class=\\"MuiInputAdornment-root MuiInputAdornment-positionEnd\\"><button class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd\\" tabindex=\\"0\\" type=\\"button\\" aria-label=\\"toggle password visibility\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78l3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z\\"></path></svg></span></button></div><fieldset aria-hidden=\\"true\\" style=\\"padding-left:8px\\" class=\\"PrivateNotchedOutline-root-156 MuiOutlinedInput-notchedOutline\\"><legend class=\\"PrivateNotchedOutline-legend-157\\" style=\\"width:0.01px\\"><span>&#8203;</span></legend></fieldset></div><p class=\\"MuiFormHelperText-root MuiFormHelperText-contained\\"></p><a class=\\"MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-linkColor-21 MuiTypography-body2 MuiTypography-colorPrimary\\" href=\\"/forgot-password\\">Forgot password?</a></div><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-2 MuiButton-containedPrimary Mui-disabled MuiButton-fullWidth Mui-disabled\\" tabindex=\\"-1\\" type=\\"submit\\" disabled=\\"\\"><span class=\\"MuiButton-label\\">Sign in</span></button><div class=\\"MuiGrid-root MuiGrid-container\\"><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12\\" style=\\"text-align:center\\">Don&#x27;t have an account? | <a class=\\"MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-linkColor-21 MuiTypography-body2 MuiTypography-colorPrimary\\" href=\\"/requestotp\\">New Registration</a></div></div></form></div></div><div class=\\"MuiBackdrop-root makeStyles-backdrop-22\\" aria-hidden=\\"true\\" style=\\"opacity:0;visibility:hidden\\"><div class=\\"MuiCircularProgress-root MuiCircularProgress-indeterminate\\" style=\\"width:40px;height:40px\\" role=\\"progressbar\\"><svg class=\\"MuiCircularProgress-svg\\" viewBox=\\"22 22 44 44\\"><circle class=\\"MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate\\" cx=\\"44\\" cy=\\"44\\" r=\\"20.2\\" fill=\\"none\\" stroke-width=\\"3.6\\"></circle></svg></div></div></div></div></div></div>"`;
